### PR TITLE
Implement pointer handling

### PR DESCRIPTION
**Pull request by andyg0808**
When using Edge on Windows 10 with a precision touchpad (PTP), Edge
[does not generate scroll events](https://blogs.windows.com/msedgedev/2017/12/07/better-precision-touchpad-experience-ptp-pointer-events). This means that zooming does not work
on PTP devices. It seems that Touch events are also not
generated. This commit implements support for the upcoming/new Pointer
API, which is [supported](https://caniuse.com/#feat=pointer) across most modern browsers. The [Pointer API](https://developer.mozilla.org/en-US/docs/Web/API/Pointer_events)
provides a unified API for pointing devices, including both mice and
touchpads.

After this pull, the Pointer API will be preferred to both the Touch
and the mouse APIs if it is present. Scroll handling will still be done when the Pointer API is in use, in order to allow the use of the mouse wheel for zoom.

We call `preventDefault` on the events in order to prevent clicks from
being generated by them in addition to the pointer events, since
they've been handled.

### CODE REVIEW

File: src/zoom.js [hunk 1]
```diff
@@ -291,6 +308,83 @@ export default function() {
...
+    p = clientPoint(this, event);
+
+    if (g.touch0 && g.touch0[2] === i) g.touch0[0] = p;
+    else if (g.touch1 && g.touch1[2] === i) g.touch1[0] = p;
+
+    t = g.that.__zoom;
+    if (g.touch1) {
+      var p0 = g.touch0[0], l0 = g.touch0[1],
+          p1 = g.touch1[0], l1 = g.touch1[1],
+          dp = (dp = p1[0] - p0[0]) * dp + (dp = p1[1] - p0[1]) * dp,
```
Comment (djmetzle) [inline src/zoom.js:351]: Woah. Inline mutation?

Comment (andyg0808) [reply src/zoom.js:351]: Yup! This code is just imitating the code from the touch version; see the existing `touch*` functions.

**Comment by jarstelfox** [comment]
CR 🌵 To be honest, matching the previous style for them is great, but that style makes it very hard to ensure this is correct.

File: src/zoom.js (hunk 1 above)
Comment (djmetzle) [reply src/zoom.js:351]: Right. Okay.

**Comment by djmetzle** [comment]
CR :raised_hand_with_fingers_splayed:

**Pull request closed without merging**